modules/account-lib/src/coin/algo/utils.ts

```diff
@@ -0,0 +1,600 @@
+   * @returns {EncodedTx} The decoded transaction.
+   */
+  decodeAlgoTxn(txnBytes: Uint8Array | string): EncodedTx {
+    let buffer: Buffer;
+    if (typeof txnBytes === 'string') {
+      if (allHexChars(txnBytes)) {
+        buffer = Buffer.from(txnBytes, 'hex');
+      } else {
+        buffer = Buffer.from(txnBytes, 'base64');
+      }
+    } else {
+      buffer = Buffer.from(txnBytes);
+    }
+
+    if (this.isDecodableUnsignedAlgoTxn(buffer)) {
+      return {
+        rawTransaction: new Uint8Array(buffer),
+        txn: algosdk.decodeUnsignedTransaction(buffer),
+        signed: false,
+      };
+    } else if (this.isDecodableSignedTransaction(buffer)) {
+      // TODO: Replace with
+      // return algosdk.Transaction.from_obj_for_encoding(algosdk.decodeSignedTransaction(buffer).txn);
+      // see: https://github.com/algorand/js-algorand-sdk/issues/364
+      // "...some parts of the codebase treat the output of Transaction.from_obj_for_encoding as EncodedTransaction.
+      // They need to be fixed(or we at least need to make it so Transaction conforms to EncodedTransaction)."
+      const tx = algosdk.decodeSignedTransaction(buffer);
+
+      const signers: string[] = [];
+      const signedBy: string[] = [];
+      if (tx.msig && tx.msig.subsig) {
+        for (const sig of tx.msig.subsig) {
+          const addr = algosdk.encodeAddress(sig.pk);
+          signers.push(addr);
+          if (sig.s) {
+            signedBy.push(addr);
+          }
+        }
+      }
+
+      return {
+        rawTransaction: new Uint8Array(buffer),
+        txn: tx.txn,
+        signed: true,
+        signers: signers,
+        signedBy: signedBy,
+      };
+    } else {
+      throw new InvalidTransactionError('Transaction cannot be decoded');
+    }
+  }
+
+  /**
+   * Estimate the transaction byte size by signing it with a random hey and calculating the byte length
+   *
+   * @param txInfo {Object} required fields to build a multisig transaction
+   * @returns {number} estimated byte size of the signed transaction
+   */
+  getTransactionByteSize(txInfo): number {
+    // The txInfo for keyreg txes only contain the object for encoding with no metadata so we can't use the normal
+    // createMultisigTransaction constructor because it will fail to construct the tx without the metadata. Since we
+    // already have the object for encoding we can instead just estimate its size directly.
+    if (!_.isUndefined(txInfo.objForEncoding)) {
+      return txInfo.estimateSize();
+    }
+
+    const transaction = this.createMultisigTransaction(Object.assign({}, { fee: 1 }, txInfo));
+    return transaction.estimateSize();
+  }
+
+  /**
+   * Create a Multisig transaction object using Algorand SDK.
+   *
+   * @param txInfo {Object} required fields to build a multisig transaction
+   * @returns {MultiSigTransaction} as defined in Algorand SDK
+   */
+  protected createMultisigTransaction(txInfo): any {
```

Comment:
I missed this the first time but we should have input/output types here (and avoid `any`). Please create a ticket.